gammapy/datasets/map.py

```diff
@@ -2321,37 +2329,298 @@ def peek(self, figsize=(12, 8)):
+            counts_data : `Map`
+                Gammapy Map containing the counts data.
+            cmap : Colormap
+                The colormap to use for plotting.
+            vmin : float
+                Minimum color limit for the log scale.
+            vmax : float
+                Maximum color limit for the log scale.
+            """
+            counts_data.plot(
+                ax=ax,
+                cmap=cmap,
+                add_cbar=True,
+                interpolation="bilinear",
+                norm=LogNorm(vmin=vmin, vmax=vmax),
+            )
+            ax.set_title("Counts")
+            ax.set_box_aspect(1)
+
+        def plot_npred(ax, npred_data, cmap, vmin, vmax):
+            """
+            Plot the npred map on a given axis.
+
+            Parameters
+            ----------
+            ax : `matplotlib.axes.Axes`
+                The matplotlib Axes to plot on.
+            npred_data : `Map`
+                Gammapy Map containing the npred data.
+            cmap : Colormap
+                The colormap to use for plotting.
+            vmin : float
+                Minimum color limit for the log scale.
+            vmax : float
+                Maximum color limit for the log scale.
+            """
+            npred_data.plot(
+                ax=ax, cmap=cmap, add_cbar=True, norm=LogNorm(vmin=vmin, vmax=vmax)
+            )
+            ax.set_title("npred")
+            ax.set_box_aspect(1)
+
+        def plot_edisp(ax, edisp_kernel):
+            """
+            Plot the energy dispersion matrix on a given axis.
+
+            Parameters
+            ----------
+            ax : `matplotlib.axes.Axes`
+                The matplotlib Axes to plot on.
+            edisp_kernel : `EDispKernel` or `EDispMap`
+                The Gammapy energy dispersion kernel or map to plot.
+
+            Returns
+            -------
+            QuadMesh
+                The resulting QuadMesh object (useful if you need a colorbar reference).
+            """
+            mesh = edisp_kernel.plot_matrix(ax=ax, add_cbar=True)
+            ax.set_xscale("log")
+            ax.set_yscale("log")
+            ax.set_title("Energy Dispersion (at FoV center)")
+            ax.set_box_aspect(1)
+            return mesh
+
+        def plot_exposure_map(ax, exposure_map, cmap):
+            """
+            Plot the exposure map at a specific energy index (e.g. middle of the energy axis).
+
+            Parameters
+            ----------
+            ax : `matplotlib.axes.Axes`
+                The matplotlib Axes to plot on.
+            exposure_map : `Map`
+                Gammapy Map containing the exposure data (with energy axis).
+            cmap : Colormap
+                The colormap to use for plotting.
+            """
+            index = int(exposure_map.geom.axes["energy_true"].nbin / 2)
+            # Dynamically scale the exposure by powers of 10 for improved readability
```

Comment:
why not just have it in log scale like the others ?

Reply:
fixed